### PR TITLE
Fix TestExecInfo failures in daint-ci

### PR DESCRIPTION
**Pull request by gronerl**
This PR changes the build mechanism of TestExecInfo so that it builds the stencil in separate folders per worker.

### CODE REVIEW

**Comment by gronerl** [comment]
bors try

**Comment by gronerl** [comment]
bors try-
bors try

**Comment by gronerl** [comment]
bors try-
bors try

**Comment by havogt** [comment]
bors try-
bors try

File: tests/cartesian_tests/unit_tests/test_exec_info.py [hunk 1]
```diff
@@ -69,9 +70,18 @@ def diffusion_def(
                 flux_x[0, 0, 0] - flux_x[-1, 0, 0] + flux_y[0, 0, 0] - flux_y[0, -1, 0]
             )
 
-    def compile_stencils(self, backend):
-        self.advection = gtscript.stencil(backend=backend, definition=self.advection_def)
-        self.diffusion = gtscript.stencil(backend=backend, definition=self.diffusion_def)
+    def compile_stencils(self, backend, worker_id):
+        modulename = inspect.currentframe().f_back.f_globals["__name__"]
```
Comment (egparedes) [inline tests/cartesian_tests/unit_tests/test_exec_info.py:74]: Just a question: is this really needed? I guess any name could be used as prefix, isnt'it?

File: tests/cartesian_tests/unit_tests/test_exec_info.py [hunk 2]
```diff
@@ -69,9 +70,18 @@ def diffusion_def(
                 flux_x[0, 0, 0] - flux_x[-1, 0, 0] + flux_y[0, 0, 0] - flux_y[0, -1, 0]
             )
 
-    def compile_stencils(self, backend):
-        self.advection = gtscript.stencil(backend=backend, definition=self.advection_def)
-        self.diffusion = gtscript.stencil(backend=backend, definition=self.diffusion_def)
+    def compile_stencils(self, backend, worker_id):
```
Comment (egparedes) [inline tests/cartesian_tests/unit_tests/test_exec_info.py:73]: We could document here , that `worker_id` comes from pytest-xdist but it's not critical

**Review by egparedes — APPROVED** [review]
It looks good, I've only have   question and a comment, both are non-blocking.

File: tests/cartesian_tests/unit_tests/test_exec_info.py (hunk 1 above)
Comment (gronerl) [reply tests/cartesian_tests/unit_tests/test_exec_info.py:74]: Sure. This gives some additional guarantee that it is unique. It is also exactly what the default is if you don't provide `name` (It's a copy paste from there.)

File: tests/cartesian_tests/unit_tests/test_exec_info.py (hunk 2 above)
Comment (gronerl) [reply tests/cartesian_tests/unit_tests/test_exec_info.py:73]: Alternatively, there is also the option to not base `name` on `worker_id` but instead just use different names for test_backcompatibility and test_accumulate.

**Pull request merged**